File: dsa/arrays/array.c
#include <stdio.h>

/*
    Arrays in C contain only one type of elements.
    Arrays in C are in the form

    (type) (arrayname)[arraysize]

        (OR)

    (type) (arrayname)[arraysize] = { element1, element2, element3}

    For example having an array of fruits would be

    char* fruits[3]

    (OR)

    char* fruits[3] = { "Mangoes", "Oranges", "Bananas"}

*/

int main()
{

    char *fruits[4] = {"Mangoes",
                       "Oranges",
                       "Avocados",
                       "Bananas"};

    int marks[5] = {40, 80, 50, 65, 20};

    // Adding or updating arrays
    marks[3] = 90;
    fruits[1] = "Strawberrys";

    printf("The marks at marks[3] is %d and the fruit at fruits[1] is %s\n", marks[3], fruits[1]);

    for (int i = 0; i < (sizeof(marks) / sizeof(int)); i++)
    {
        printf("The marks at marks[%d] is %d\n", i, marks[i]);
    }

    for (int i = 0; i < (sizeof(fruits) / sizeof(char *)); i++)
    {
        printf("The fruit at fruits[%d] is %s\n", i, fruits[i]);
    }
}

File: dsa/arrays/array.js
/*


*/

var marks = [40, 80, 50, 90, 20]

var fruits = [
    "Mangoes",
    "Oranges",
    "Avocados",
    "Bananas"
];

for (mark = 0; mark < marks.length; mark++)
    console.log(`${marks[mark]}`)

console.log("\n")

for (fruit = 0; fruit < fruits.length; fruit++)
    console.log(`${fruits[fruit]}`)

console.log("\n")


// Array methods in JavaScript

console.log("\n**Map**")
fruits.map((fruit) => {
    console.log(fruit)
})


fruits.push("Watermelon")
var filtered = fruits.filter((element) => element == 'Mangoes')

console.log("\n**Filtered data**\n", filtered)

console.log("\n**ForEach**")
fruits.forEach(fruit => { console.log(fruit) })
